dotnet/src/IntegrationTests/Connectors/Memory/Pinecone/PineconeVectorStoreFixture.cs

```diff
@@ -0,0 +1,306 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Connectors.Pinecone;
+using Microsoft.SemanticKernel.Data;
+using Pinecone.Grpc;
+using Xunit;
+using Sdk = Pinecone;
+
+namespace SemanticKernel.IntegrationTests.Connectors.Memory.Pinecone;
+
+public class PineconeVectorStoreFixture : IAsyncLifetime
+{
+    private const int MaxAttemptCount = 100;
+    private const int DelayInterval = 300;
+    public const string IndexName = "sk-index";
```

Comment:
I did a fairly crude attempt at avoiding conflicts where different people or builds are running the same integration tests for the Azure AI Search integration tests.  I'm appending a sanitized version of the machine name onto the index name:
`new Regex("[^a-zA-Z0-9]").Replace(Environment.MachineName.ToLowerInvariant(), "")`.
Since this is also potentially relying on a single instance, it'll be nice to do the same here.

Reply:
sounds good, added the same. One small change is that I added timeout for security reasons. This shouldn't really matter but can never be too safe ;) In EF we basically have a blanket rule that all regex must have a timeout (since IIRC we ran into DoS troubles with those in the past)